Sources/TuistKit/Services/InitService.swift

```diff
@@ -1,266 +1,207 @@
-            }
-        }
+        return nil
     }
 
-    /// Finds template directory
-    /// - Parameters:
-    ///     - templateDirectories: Paths of available templates
-    ///     - template: Name of template
-    /// - Returns: `AbsolutePath` of template directory
-    private func templateDirectory(templateDirectories: [AbsolutePath], template: String) throws -> AbsolutePath {
-        guard let templateDirectory = templateDirectories.first(where: { $0.basename == template })
-        else { throw InitServiceError.templateNotFound(template) }
-        return templateDirectory
-    }
-
-    /// Returns name to use. If `name` is nil, returns a directory name executed `init` command.
-    private func name(_ name: String?, path: AbsolutePath) throws -> String {
-        if let name {
-            return name
-        } else if let directoryName = path.components.last {
-            return directoryName
-        } else {
-            throw InitServiceError.ungettableProjectName(AbsolutePath.current)
-        }
-    }
-
-    private func path(_ path: String?) throws -> AbsolutePath {
-        if let path {
-            return try AbsolutePath(validating: path, relativeTo: FileHandler.shared.currentPath)
-        } else {
-            return FileHandler.shared.currentPath
-        }
+    private func nameOfXcodeProjectOrWorkspace(in directory: AbsolutePath) async throws -> StartPromptingWorkflowType {
+        let xcodeProjectsAndWorkspaces = try await findXcodeProjectsAndWorkspaces(in: directory)
+        return prompter
+            .promptWorkflowType(
+                xcodeProjectOrWorkspace: xcodeProjectsAndWorkspaces
+                    .first(where: \.isWorkspace) ?? xcodeProjectsAndWorkspaces.first(where: \.isProject)
+            )
     }
 
-    private func platform(_ platform: String?) throws -> Platform {
-        if let platformString = platform {
-            if let platform = Platform(rawValue: platformString) {
-                return platform
-            } else {
-                throw InitServiceError.invalidValue(argument: "platform", error: "Platform should be either ios, tvos, or macos")
-            }
-        } else {
-            return .iOS
-        }
+    private func findXcodeProjectsAndWorkspaces(in directory: AbsolutePath) async throws -> Set<XcodeProjectOrWorkspace> {
+        var paths = Set(
+            try await fileSystem.glob(directory: directory, include: ["**/*.xcworkspace"]).collect()
+                .map(XcodeProjectOrWorkspace.workspace)
+        )
+        paths
+            .formUnion(
+                try await fileSystem.glob(directory: directory, include: ["**/*.xcodeproj"]).collect()
+                    .map(XcodeProjectOrWorkspace.project)
+            )
+        paths
+            .subtract(
+                try await fileSystem.glob(directory: directory, include: ["**/*.xcodeproj/project.xcworkspace"]).collect()
+                    .map(XcodeProjectOrWorkspace.workspace)
+            )
```

Comment:
Instead of globbing again which can be an expensive operation, you can filter out workspace paths where the parent directory's extension is `.xcodeproj`.